Bottom: 761e7f98132819c1ee957b805412335e6c31a231
Top:    de128f5c6219563168aa8c1ffeda6298221c7778
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-15 18:58:54 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index e92cb81..f36b465 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2520,7 +2520,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
